examples/Tutorial 5 - Graph Coloring Solver.ipynb

```diff
@@ -0,0 +1,306 @@
+   "id": "a29ee1bf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "instance.draw()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f209d04d",
+   "metadata": {},
+   "source": [
+    "## Solving the Graph Coloring Problem\n",
+    "\n",
+    "We will use the greedy heuristic algorithm described in [Appendix A](https://arxiv.org/pdf/2301.02637) to find a coloring of the graph using MIS output. \n",
+    "\n",
+    "The algorithm starts with a set $S$ of all the nodes in the graph, and at each iteration it searches for a maximum independent set of nodes of the subgraph formed by the nodes currently in $S$, colors all of the nodes of the MIS in the same color, then removes them from $S$. The operation is repeated until $S$ is empty.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "525505d4",
+   "metadata": {},
+   "source": [
+    "### Using a classical solver\n",
+    "We will first solve the coloring problem using the standard classical and heuristic [MIS solver in Networkx](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.approximation.clique.maximum_independent_set.html). As it is heuristic and non-deterministic, this solver does not guarantee an optimal solution."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cb3177f5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "solver = GraphColoringSolver(loader, 1.2, SolverConfig(\n",
+    "    method = MethodType.EAGER,\n",
+    "    max_iterations=1\n",
+    "))\n",
+    "solver.solve()\n",
+    "solver.visualize_solution()\n",
+    "print(solver.colors)   \n",
+    "print(f\"Number of colors used: {solver.colors_count}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d753e7ee",
+   "metadata": {},
+   "source": [
+    "The array `solver.colors` represents the group assigned to each antenna by the algorithm, such that all the antennas of the same group can share the same frequency without interfering with each other."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "de0f984a",
+   "metadata": {},
+   "source": [
+    "### Using the quantum SDK QuTiP\n",
+    "We will now use a quantum solver to solve the MIS instances used by our coloring algorithm, please refer to tutorial 1a for more details about the solver. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "334e1a05",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "solver = GraphColoringSolver(loader, 1.2, SolverConfig(\n",
+    "    method = MethodType.EAGER,\n",
+    "    backend = QutipBackend(),\n",
+    "    max_iterations=1\n",
+    "))\n",
+    "solver.solve()\n",
+    "solver.visualize_solution()\n",
+    "print(solver.colors)   \n",
+    "print(f\"Number of colors used: {solver.colors_count}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b4e819e",
+   "metadata": {},
+   "source": [
+    "## Enhancing our algorithm\n",
```

Comment:
Sorry, maybe I was a bit confusing here. What I meant was to use the existing pre- and post-processing methods to see if that improves the overall result quality. You can find an example on how to use them in the tests for example [here](https://github.com/pasqal-io/maximum-independent-set/blob/f0620df0dbc2f9e8e8f54c3a38e4ac950000e992/tests/test_classical_mis.py#L33).

Reply:
Alright, could you add them nevertheless ? Just as illustrative on how to use pre- and post-processing in a real-life example. Thanks.